Habitica/src/com/habitrpg/android/habitica/ui/activities/TaskFormActivity.java

```diff
@@ -510,7 +600,7 @@ private void prepareSave() {
             this.task.save();
             new Select()
                     .from(Tag.class)
-                    .where(Condition.column("id").in("", tags.toArray())).async().queryList(tagsSearchingListener);
+                    .where(Condition.column("id").in("", userSelectedTagIds.toArray())).async().queryList(tagsSearchingListener);
```

Comment:
Only add the Tags the user specified.

tags = List of all tags
userSelectedTagsId = list of tags user selected